node/tests/SharedTests.ts

```diff
@@ -8584,6 +8585,65 @@ export function runBaseTests<Context>(config: {
         },
         config.timeout,
     );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "check that blocking commands never time out %p",
+        async (protocol) => {
+            await runTest(async (client: BaseClient, cluster) => {
+                const key1 = "{blocking}-1-" + uuidv4();
+                const key2 = "{blocking}-1-" + uuidv4();
+                const keyz = [key1, key2];
+
+                // TODO: xreadgroup
+                const promiseList: Promise<ReturnType>[] = [
+                    client.bzpopmax(keyz, 0),
+                    client.bzpopmin(keyz, 0),
+                    client.blpop(keyz, 0),
+                    client.brpop(keyz, 0),
+                    client.xread(
+                        { [key1]: "0-0", [key2]: "0-0" },
+                        { block: 0 },
+                    ),
+                    client.wait(42, 0),
+                ];
+
+                if (!cluster.checkIfServerVersionLessThan("6.2.0")) {
+                    promiseList.push(
+                        client.blmove(
+                            key1,
+                            key2,
+                            ListDirection.LEFT,
+                            ListDirection.LEFT,
+                            0,
+                        ),
+                    );
+                }
+
+                if (!cluster.checkIfServerVersionLessThan("7.0.0")) {
+                    promiseList.push(
+                        client.blmpop(keyz, ListDirection.LEFT, 0),
+                        client.bzmpop(keyz, ScoreFilter.MAX, 0),
+                    );
+                }
+
+                try {
+                    for (const promise of promiseList) {
+                        const timeoutPromise = new Promise((resolve) => {
+                            setTimeout(resolve, 500);
+                        });
+                        await Promise.race([promise, timeoutPromise]);
```

Comment:
Not sure if I understand the test, how do you know which one has resolved? 
If I understand, what the test does is checking if at least one of the promises has resolved.
Then you resolve all the promises, if they are blocked and never timed out, we expect it to never resolve right?
So how the test doing it?

Reply:
Updated